Guides for OCI and AWS Metrics (#1042)
micronaut-projects/micronaut-guides: guides/micronaut-metrics-oci/micronaut-metrics-oci.adoc

external:micronaut-metrics/start.adoc[]

common:oracle-cloud-account.adoc[]

external:micronaut-metrics/create-app.adoc[]

== Monitor metrics on Oracle Cloud

> For publishing metrics on the Oracle Cloud you have to adjust your application configuration slightly. With that configuration change you should be able to create queries and monitor your metrics on the Oracle Cloud https://cloud.oracle.com/monitoring/explore[Metrics Explorer].

Reviewer: ```suggestion
To publish metrics to Oracle Cloud, you have to adjust your application configuration slightly, and then you will be able to create queries and monitor your metrics on the Oracle Cloud https://cloud.oracle.com/monitoring/explore[Metrics Explorer].
```

resource:application.yml[tag=metrics]

<1> Change to `false` to disable all metrics, e.g., per-environment

> <2> These default to `true` and are only here for convenience, to be able to disable a subset of metrics

Reviewer 2: ```suggestion
<2> These binders default to `true`. We list them for convenience to encourage you to experiment with disabling some of them. 
```

You could extract a callout and reuse it between both guides.

> <3> Change namespace to something meaningful, for an example app name. In this guide we are using `micronaut_guide`

Reviewer 2: you can extract a callout and reuse it in both guides.

[NOTE]
====
By default, metrics will be published in your root compartment (tenancy). If you want to specify in which compartment metrics should be published, you can add `compartmentId` field inside your configuration. The full path of the key is `micronaut.metrics.export.oraclecloud.compartmentId`.
====

=== Visit the Oracle Cloud https://cloud.oracle.com/monitoring/explore[Metrics Explorer].

> In the query editor create the query that you want to monitor. You have to specify the compartment (#1), namespace (#2) and the name of the metric (#3).

Reviewer 2: ```suggestion
In the query editor, create the query that you want to monitor. You have to specify the compartment (#1), namespace (#2), and the name of the metric (#3).
```

image::oci-metrics/query-editor.png[]

> After clicking the (#4) Update Chart button you will see values of your selected metric in the Metrics Explorer.

Reviewer 2: ```suggestion
After clicking the (#4) Update Chart button, you will see the values of your selected metric in the Metrics Explorer.
```

> After clicking on the (#4) Update Chart button you should be able to see values of your selected metric in the Metrics Explorer.

Reviewer: ```suggestion
After clicking the (#4) Update Chart button you will see values of your selected metric in the Metrics Explorer.
```
